libs/core-components/src/lib/Button/Button.tsx

```diff
@@ -102,9 +102,7 @@ const Button: React.FC<ButtonProps> = ({
       ) : (
         ''
       )}
-      <span className="c-button__text" data-testid="text">
-        {children}
-      </span>
+      {children}
```

Comment:
Removes excess markup (that definitely causes minor excess space, and can cause minor alignment issues).